internal/controllers/clusterextension_registryv1_validation_test.go

```diff
@@ -116,6 +117,7 @@ func TestClusterExtensionRegistryV1DisallowDependencies(t *testing.T) {
 				ActionClientGetter:    helmClientGetter,
 				Unpacker:              unpacker,
 				InstalledBundleGetter: mockInstalledBundleGetter,
+				Finalizers:            crfinalizer.NewFinalizers(),
```

Comment:
It's a bit disconcerting to me that so much of the `ClusterExtensionReconciler` can be constructed _so_ differently than the real one can. We should discuss further, but off the top of my head, I wonder if we can use the exact same reconciler as is configured in `main.go`.
1. For catalogd we can use httptest to setup a mock catalogd server and use the _real_ ClusterExtension http client.
2. An image registry is an http server. I wonder if there is an in-process implementation that would be easy to run for test purposes.
3. Use `t.TempDir()` subdirectories for any necessary local storage and caching that happens during reconcile. 
4. For the kubernetes cluster, we use envtest.

If we had that setup, we wouldn't actually need to configure the ClusterExtension reconciler any differently.

Reply:
More immediately, we should have a unit tests that make sure finalizers are actually being handled properly during reconcile.

Reply:
> It's a bit disconcerting to me that so much of the ClusterExtensionReconciler can be constructed so differently than the real one can.

This is definitely up for discussion and needs separate set of effort. The whole point of these unit tests is to make sure that the reconciler code path traversed during various instances is as expected. Which is why we ensure that the mock unpacker returns quickly as expected. The code coverage for unpacking and storage has separate unit tests (currently in rukpak).

I'm not sure if we immediately need to focus on this - rather if possible try finishing up https://github.com/operator-framework/operator-controller/issues/879 as immediately as possible (even if it requires us to mock some of the components) to ensure we don't skip any nuances in the reconciler code.